modules/class.jetpack-calypsoify.php

```diff
@@ -0,0 +1,195 @@
+<?php
+/*
+ * This is Calypso skin of the wp-admin interface that is conditionally triggered via the ?calypsoify=1 param.
+ * Portted from an internal Automattic plugin.
+*/
+
+class Jetpack_Calypsoify {
+	static $instance = false;
+
+	private function __construct() {
+		add_action( 'wp_loaded', array( $this, 'setup' ) );
+	}
+
+	public static function getInstance() {
+		if ( ! self::$instance ) {
+			self::$instance = new self();
+		}
+
+		return self::$instance;
+	}
+
+	public function setup() {
+		add_action( 'admin_init', array( $this, 'check_param' ) );
+		if ( 1 == (int) get_user_meta( get_current_user_id(), 'calypsoify', true ) ) {
```

Comment:
Could we introduce a filter for forcing calypsoify to display with options other than then user meta? I'm thinking of specifically being able to target certain store AT sites.

I don't think @timmyc and I have fully figured out how we want to determine to 'calyposify' Woo -- but it's possible we might want to do something like set a site site option rather then rely on a query parameter. Otherwise store users could get confused with the layout getting set and unset for them.

Reply:
Would the site option be somehow tied to the user that initiated it? Otherwise, secondary users of the site would also see it Calypsoified if the site option was set which I think would be even more confusing. This is meant to be a single flow type of thing for people 
- hits Calypso link 
- sees wp-admin Calypsoified 
- hits "go back" 

If the user never goes back to wp-admin, any secondary user of the site would see it Calypsoified.

Reply:
Adding a filter is a good idea anyway